src/middlewared/middlewared/plugins/device_/vrrp_events_linux.py

```diff
@@ -0,0 +1,57 @@
+import os
+import time
+
+from middlewared.utils import start_daemon_thread
+
+VRRP_FIFO_FILE = '/var/run/vrrpd.fifo'
+TIMEOUT = 2
+VRRP_FIFO_CONNECTED = False
+
+
+def vrrp_hook_license_update(middleware, prev_product_type, *args, **kwargs):
+
+    # get new product_type
+    new_product_type = middleware.call_sync('system.product_type')
+
+    if prev_product_type != 'SCALE_ENTERPRISE':
+        if new_product_type == 'SCALE_ENTERPRISE':
+            vrrp_fifo_listen(middleware)
+
+
+def vrrp_fifo_listen(middleware):
+
+    # global var used to ensure we don't start more than 1 thread
+    # listening on the FIFO.
+    global VRRP_FIFO_CONNECTED
+
+    if not VRRP_FIFO_CONNECTED:
+
+        # create the fifo, ignoring if it already exists
+        try:
+            os.mkfifo(VRRP_FIFO_FILE)
+        except FileExistsError:
+            pass
+
+        while True:
+            try:
+                with open(VRRP_FIFO_FILE) as f:
+                    VRRP_FIFO_CONNECTED = True
+                    middleware.logger.info('vrrp fifo connection established')
+                    # all vrrp messages are terminated with a newline
+                    for line in f:
+                        middleware.call_hook_sync('vrrp.fifo', data=line)
+            except Exception:
+                middleware.logger.error('vrrp fifo connection not established, retrying')
+                # sleep for `TIMEOUT` before trying to open FIFO and send event again
+                time.sleep(TIMEOUT)
+
+
+def setup(middleware):
+
+    # register hook to be called if/when a license has been uploaded
+    # to system changing the product type to 'SCALE_ENTERPRISE'
+    middleware.register_hook('vrrp.post_license_update', vrrp_hook_license_update, sync=False)
```

Comment:
Sorry, I was meaning used the same existing hook, just implemented the call here, not a new hook.